modules/local/subworkflow/mapping.nf

```diff
@@ -0,0 +1,136 @@
+/*
+================================================================================
+                                     MAPPING
+================================================================================
+*/
+
+include { BWAMEM2_MEM }            from '../process/bwamem2_mem'
+include { BWA_MEM as BWAMEM1_MEM } from '../process/bwa_mem'
+include { MERGE_BAM }              from '../process/merge_bam'
+include { QUALIMAP_BAMQC }         from '../../nf-core/software/qualimap_bamqc'
+include { SAMTOOLS_INDEX }         from '../../nf-core/software/samtools/index'
+include { SAMTOOLS_STATS }         from '../../nf-core/software/samtools/stats'
+
+workflow MAPPING {
+    take:
+        step            //   value: [mandatory] starting step
+        input_sample    // channel: [mandatory] input_sample
+        target_bed      // channel: [optional]  target_bed
+        bwa             // channel: [mandatory] bwa
+        fasta           // channel: [mandatory] fasta
+        fai             // channel: [mandatory] fai
+        samtools_opts   //     map: options for SAMTOOLS_INDEX module
+        merge_bam_opts  //     map: options for MERGE_BAM module
+        skip_bamqc      // boolean: true/false
+        skip_samtools   // boolean: true/false
+        save_bam_mapped // boolean: true/false
+
+    main:
+
+    bam_mapped_indexed = Channel.empty()
+    bam_reports        = Channel.empty()
+
+    if (step == "mapping") {
+        reads_input = input_sample
+
+        bam_bwamem1 = Channel.empty()
+        bam_bwamem2 = Channel.empty()
+
+        if (params.aligner == "bwa-mem") {
+            BWAMEM1_MEM(reads_input, bwa, fasta, fai, params.modules['bwa_mem'])
+            bam_bwamem1 = BWAMEM1_MEM.out.bam
+        } else {
+            BWAMEM2_MEM(reads_input, bwa, fasta, fai, params.modules['bwamem2_mem'])
+            bam_bwamem2 = BWAMEM2_MEM.out
+        }
+
+        bam_bwa = bam_bwamem1.mix(bam_bwamem2)
+
+        bam_bwa.map{ meta, bam ->
+            patient = meta.patient
+            sample  = meta.sample
+            gender  = meta.gender
+            status  = meta.status
+            [patient, sample, gender, status, bam]
+        }.groupTuple(by: [0,1])
+            .branch{
+                single:   it[4].size() == 1
+                multiple: it[4].size() > 1
+            }.set{ bam_bwa_to_sort }
+
+        bam_bwa_single = bam_bwa_to_sort.single.map {
```

Comment:
This remains empty due to the branching above, when multiple bams per sample are provided, right?

Reply:
yes unless a single pair of fastq for a sample is provided. It's empty.

Reply:
The idea is to be able to manage a mixture of single pairs and multiple pairs of sample in just one go.

Reply:
That is what I though :) Just wanted to make sure that it really stays empty, otherwise the mix operation could make a file end up twice in there